chain-abci/src/app/app_init.rs

```diff
@@ -8,36 +9,43 @@ use chain_core::init::config::InitConfig;
 use chain_core::state::{BlockHeight, RewardsPoolState};
 use chain_core::tx::{
     data::{attribute::TxAttributes, Tx, TxId},
+    fee::LinearFee,
     TxAux,
 };
 use hex::decode;
-use integer_encoding::VarInt;
-use log::info;
+use log::{info, warn};
 use protobuf::Message;
-use rlp::{Decodable, Encodable, Rlp, RlpStream};
+use rlp::{Encodable, RlpStream};
+use serde::{Deserialize, Serialize};
 
 use crate::storage::*;
 
+#[derive(Serialize, Deserialize, PartialEq, Debug, Clone)]
+pub struct ChainNodeState {
```

Comment:
I think the idea of declaring `FeeAlgorithm` trait is to enable easy switching of `fee_policy` in future. But, current design will not allow that easy switching. Changing `ChainNodeState` in following way will help:
```
pub struct ChainNodeState<T> where T: FeeAlgorithm {
    // ... other fields ... //

    pub fee_policy: T
}
```

Reply:
it'll be a bit of a rabbit hole (as this gets transitively into almost everywhere in ABCI), but it should be better to do now than later

Reply:
added a TODO comment there, as the scope of changing this is quite large